fail2ban/server/filter.py

```diff
@@ -606,22 +618,59 @@ def processLine(self, line, date=None):
 				tupleLine = (line[:s], m, line[e:])
 				if m: # found and not empty - retrive date:
 					date = self.dateDetector.getTime(m, timeMatch)
-
-				if date is None:
-					if m: logSys.error("findFailure failed to parse timeText: %s", m)
+					if date is not None:
+						# Lets get the time part
+						date = date[0]
+						self.__lastTimeText = m
+						self.__lastDate = date
+					else:
+						logSys.error("findFailure failed to parse timeText: %s", m)
+				# matched empty value - date is optional or not available - set it to last known or now:
+				elif self.__lastDate and self.__lastDate > MyTime.time() - 60:
+					# set it to last known:
+					tupleLine = ("", self.__lastTimeText, line)
 					date = self.__lastDate
 				else:
-					# Lets get the time part
-					date = date[0]
-					self.__lastTimeText = m
+					# set it to now:
+					date = MyTime.time()
+			else:
+				tupleLine = ("", "", line)
+			# still no date - try to use last known:
+			if date is None:
+				noDate = True
+				if self.__lastDate and self.__lastDate > MyTime.time() - 60:
+					tupleLine = ("", self.__lastTimeText, line)
+					date = self.__lastDate
+		
+		if self.checkFindTime:
+			# if in operation (modifications have been really found):
+			if self.inOperation:
+				# if weird date - we'd simulate now for timeing issue (too large deviation from now):
+				if (date is None or date < MyTime.time() - 60 or date > MyTime.time() + 60):
```

Comment:
Forgive my late comment. What unit does this 60 represent? Seconds? If yes, then how can 60 seconds have anything to do with timezone?

On Ubuntu 22.04 we now have fail2ban 0.11.2 (with this change), and suddenly these warnings pop up. They pop up if a message is a little over one minute later than expected.

Reply:
> What unit does this 60 represent? Seconds? 

Yes.

> If yes, then how can 60 seconds have anything to do with timezone?

It doesn't, it has something with too large deviation (so fail2ban in operation mode starts to use NOW instead of read datepattern in order to handle failure and banning intervals as correct as possible). Such a long delay may be a TZ issue (and more that 90% of users had that), but this warning is rather a signal that something is not alright with the jail/filter/service.

As for confusing warnings - see #2882, and it was changed in #3053.

Reply:
Thanks for the explanation